PSKoans/Koans/Cmdlets 1/AboutSelectObject.Koans.ps1

```diff
@@ -4,69 +4,295 @@ param()
-        __ | Should -Be $Object.Length
+        $Selected.Attributes | Should -BeNullOrEmpty
     }
 
     It 'changes the object type' {
-        $FileObject = Get-Item -Path $home
+        <#
+            When properties are selected from an object a new custom object is created.
+
+            The new object is also tagged with a PSTypeName derived from the original type name.
+        #>
+
+        $Folder = Get-Item -Path $PSHome
 
-        $FileObject | Should -BeOfType __
+        $Selected.GetType().FullName | Should -BeOfType [System.IO.Directoryinfo]
 
-        $Object = $FileObject | Select-Object -Property FullName, Name, DirectoryName
-        $Object | Should -BeOfType __
+        'System.IO.Directoryinfo' | Should -BeIn $Folder.PSTypeNames
+
+        # The new selected object is an instance of [System.Management.Automation.PSCustomObject].
+
+        $Selected = $Folder | Select-Object -Property * -ExcludeProperty Attributes
+
+        $Selected | Should -BeOfType [System.Management.Automation.PSCustomObject]
+
+        '___' | Should -BeIn $Selected.PSTypeNames
     }
 
     It 'can retrieve just the contents or value of a property' {
-        $FileObject = Get-Item -Path $home
+        # Individual properties can be expanded, retrieving the just a value.
+
+        $PropertyToExpand = '___'
 
-        $FileName = $FileObject | Select-Object -ExpandProperty __ -ErrorAction SilentlyContinue
+        $Value = Get-Item -Path $PSHome | Select-Object -ExpandProperty $PropertyToExpand
 
-        $FileName | Should -Be $FileObject.Name
+        $Value | Should -Be 'Directory'
+    }
+
+    It 'can merge the properties of a nested object with properties from the parent' {
+        <#
+            The ExpandProperty parameter can be used to "move" the properties of a nested
+            property up to a new parent object.
+        #>
+
+        $PowerShellExe = Get-Process -Id $PID | Select-Object -ExpandProperty Path | Get-Item
+
+        $Selected = $PowerShellExe | Select-Object Name -ExpandProperty VersionInfo
+
+        # The resulting object will contain all of the properties found under VersionInfo.
+
+        $Selected.____ | Should -Be $PowerShellExe.FullName
     }
 
     It 'can pick specific numbers of objects' {
         $Array = 1..100
 
         $FirstThreeValues = $Array | Select-Object -First 3
-        __ | Should -Be $FirstThreeValues
+        @(__, __, __) | Should -Be $FirstThreeValues
 
         $LastFourValues = $Array | Select-Object -Last 4
-        __ | Should -Be $LastFourValues
+        @(__, __, __, __) | Should -Be $LastFourValues
 
         $Values = $Array | Select-Object -Skip 10 -First 5
-        __ | Should -Be $Values
+        @(__, __, __, __, __) | Should -Be $Values
+
+        # SkipLast cannot be used alongside the Last, First, and Skip parameters.
+
+        $Values = $Array | Select-Object -SkipLast 95
+        @(__, __, __, __, __) | Should -Be $Values
```

Comment:
This won't parse, unfortunately. I think the best we can do here is define the initial array as `$Array = 1..100 -as [string[]]` and then have all these blanks in quotes.